Features/ft 835 merge hd export

roberthorlings opened the pull request:
Merge of rc2_dev export code into master branch, to enable exporting using the core-api

roberthorlings committed:
Merged export controller and services from rc2_dev branch

Also split the exportService into a metadata part and an export part.
However, no extra changes or improvements were made to make the export
work. Just the code is merged.

roberthorlings committed:
Merged frontend code for exporting from rc2_dev

roberthorlings committed:
Merging changes in dataExportService from rc2_dev to use core-api for
exporting

roberthorlings committed:
Merged even more export code from the rc2_dev branch to make exporting
work.

cataphract commented:
This has direct references to core-db (packages `org.transmartproject.db.*`). core-db is a runtime dependency, even if Grails doesn't enforce the compile/runtime separation. Refactor these out, either by
1. Using core-api exposed functionality. Core-api classes are in `org.transmartproject.core.*`.
2. If functionality is not available there (or it's too inefficient), you may need to add it to core-api.
3. As a last resort, move the functionality to transmartApp. This should be avoided; we want to route as much functionality through core-api as possibly so that we can for instance enforce security constraints centrally.

cataphract commented:
Reorganize imports. A lot of them are clearly unnecessary just by looking at the diff (for instance there are imports to `java.util.*` classes).

roberthorlings committed:
Removed references to the core-db, as well as unused imports

cataphract commented:
`DataCountService` still has references to core-db.

JanKanis committed:
Remove 'created by ...' comment

We decided not to include those

roberthorlings committed:
Removed reference to core-db in export functionality.

This commit depends on changes in core-db (commit 930b474) and core-api
(commit 8906be9).

roberthorlings committed:
Reimplemented counting patients with clinical data

roberthorlings committed:
Adjusted export to also handle non BiomakerDataRow objects

cataphract commented:
Can you merge with master? Seems that could be the reason the travis build is failing.

roberthorlings commented:
I will. Another reason that travis is failng is that it used a branch from transmart-core-api which contained a method that was not implemented. but I'll merge with master

roberthorlings committed:
Merge remote-tracking branch 'origin/master' into features/FT-835-merge-hd-export

cataphract commented:
I still see an import of a core-db class and java.util classes.

roberthorlings committed:
Forgot to organize imports

cataphract commented on grails-app/services/com/recomdata/asynchronous/GenericJobService.groovy:
```diff
@@ -51,16 +43,16 @@ import org.rosuda.REngine.Rserve.*;
+    def i2b2ExportHelperService = ctx.i2b2ExportHelperService
+    def snpDataService = ctx.snpDataService
+    def dataExportService = ctx.dataExportService
+    def asyncJobService = ctx.asyncJobService
```
What happened here? Why is injection not used anymore?

cataphract commented on grails-app/services/com/recomdata/asynchronous/GenericJobService.groovy:
```diff
@@ -235,45 +226,36 @@ class GenericJobService implements Job {
+                            }
                         }
+                    } catch (Exception e) {
+                        println("Failed to FTP PUT the ZIP file");
```
println

cataphract commented on grails-app/services/com/recomdata/transmart/data/export/DataExportService.groovy:
```diff
@@ -100,14 +110,32 @@ class DataExportService {
+                        switch (selectedFile) {
 							case "STUDY":
 								retVal = metadataService.getData(studyDir, "experimentalDesign.txt", jobDataMap.get("jobName"), studyList);
+                                log.info("retrieved study data")								
```
This is at most a debug statement, trace or no statement at all would be even better..

roberthorlings committed:
Moved GenericJobService to src as it is not a service
